### ISSUE TITLE
Ctrl+f searching for block text should cause matches to scroll into view

### ISSUE THREAD

**Issue by jwzimmer** [LLK/scratch-gui#764]
### Expected Behavior

When you find a block by searching the page for text on the block, you should be able to use that functionality to navigate to the matching blocks.

### Actual Behavior

@fsih reported on 10/11/17:
> When I ctrl+f to find a block it should scroll into view

### Steps to Reproduce

Try to ctrl+f (or command+f) to find a block. You can see that the text in the block is found/ searched appropriately if you search for a term in a visible block. Try searching for a block that is off-screen/ out of your view - note that there is a result indicated but the block is not scrolled to.

### Operating System and Browser

_Chrome, Mac (Sri Lanka machine), at https://ericrosenbaum.github.io/scratch-gui/extensions-oct2017/_

**Comment by amazinigmech2418** [comment]
Maybe, an override of the normal ctrl. + f is needed so that it will create a new find feature (that probably already exists for it to read inside canvas) and it needs to change 2 variables (scroll Y and scroll X) to the position of the word. Also, the variables would need to be created if not already and cause a scroll effect inside the editor. I hope this helps!

**Comment by hedgehog125** [comment]
I agree with @amazinigmech2418, but what if when you search it could bring up a small block editor and then you could create a script there and it could look for where the blocks occur. This would be much easier for new scratchers and could be a lot more useful.

**Comment by amazinigmech2418** [comment]
YES!!!!!!!!!!! That's a great idea! It could convert it to JSON and find it in the JSON code for the blocks and find the exact position of the script!

**Comment by mrjacobbloom** [comment]
There's no JS api to detect when a browser pulls up the find through a context menu, or when a user navigates to the next/prev item, although it could trigger a focus event (??). As suggested it's probably easiest to implement the whole thing from the ground up

**Comment by hedgehog125** [comment]
Maybe there could be a button instead, because otherwise it wouldn't work on tablets.

**Comment by Kenny2github** [comment]
The point is ctrl+f currently uses the builtin browser find in page thingy - so the new handler would override that shortcut. I do think that there could be a button to do this, and then ctrl+f would trigger that button, instead of having the key handler handle that.

**Comment by amazinigmech2418** [comment]
That would work way better actually. I've seen some websites before that change what the shortcut does, but don't know how they do it, but once the method to trigger the search is finished, we can add the code for the searching.

**Comment by griffpatch** [comment]
This feature is urgently needed for any substantially sized project. The only way I can find things at present is to do the Ctrl+F, type in a search string, then zoom out and try to pinpoint the yellow highlighted text in the tiny blocks on screen while scrolling around. Just having it scroll into view would be such a relief! I agree that overriding the Ctrl+F like google apps tend to do would be a good plan, as long as the search does achieve something similar though, because it could actually be really useful to be able to find variable names, usages, custom blocks, string literals, parameters, comments, etc...
